components/content-service/pkg/storage/s3.go

```diff
@@ -0,0 +1,417 @@
+	})
+
+	var nsk *types.NoSuchKey
+	if errors.As(err, &nsk) {
+		return nil, ErrNotFound
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := rs.PresignedFactory().PresignGetObject(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(rs.Config.Bucket),
+		Key:    aws.String(obj),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &DownloadInfo{
+		Meta: ObjectMeta{
+			// TODO(cw): implement this if we need to support FWB with S3
+		},
+		Size: resp.ObjectSize,
+		URL:  req.URL,
+	}, nil
+}
+
+// SignUpload implements PresignedAccess
+func (rs *PresignedS3Storage) SignUpload(ctx context.Context, bucket string, obj string, options *SignedURLOptions) (info *UploadInfo, err error) {
+	resp, err := rs.PresignedFactory().PresignPutObject(ctx, &s3.PutObjectInput{
+		Bucket: &rs.Config.Bucket,
+		Key:    aws.String(obj),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &UploadInfo{
+		URL: resp.URL,
+	}, nil
+}
+
+func newDirectS3Access(client S3Client, config S3Config) *s3Storage {
+	return &s3Storage{
+		Config: config,
+		client: client,
+	}
+}
+
+type s3Storage struct {
+	Config S3Config
+
+	OwnerID, WorkspaceID, InstanceID string
+
+	client S3Client
+}
+
+// Bucket implements DirectAccess
+func (s3st *s3Storage) Bucket(userID string) string {
+	return s3st.Config.Bucket
+}
+
+// BackupObject implements DirectAccess
+func (s3st *s3Storage) BackupObject(name string) string {
+	return s3st.objectName(name)
+}
+
+// Download implements DirectAccess
+func (s3st *s3Storage) Download(ctx context.Context, destination string, name string, mappings []archive.IDMapping) (found bool, err error) {
+	return s3st.download(ctx, destination, s3st.objectName(name), mappings)
+}
+
+// DownloadSnapshot implements DirectAccess
+func (s3st *s3Storage) DownloadSnapshot(ctx context.Context, destination string, name string, mappings []archive.IDMapping) (found bool, err error) {
+	return s3st.download(ctx, destination, name, mappings)
+}
+
+func (s3st *s3Storage) download(ctx context.Context, destination string, obj string, mappings []archive.IDMapping) (found bool, err error) {
+	resp, err := s3st.client.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(s3st.Config.Bucket),
+		Key:    aws.String(obj),
+	})
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	err = archive.ExtractTarbal(ctx, resp.Body, destination, archive.WithUIDMapping(mappings), archive.WithGIDMapping(mappings))
+	if err != nil {
+		return true, xerrors.Errorf("tar %s: %s", destination, err.Error())
+	}
+
+	return true, nil
+}
+
+// EnsureExists implements DirectAccess
+func (*s3Storage) EnsureExists(ctx context.Context) error {
+	return nil
+}
+
+// Init implements DirectAccess
+func (s3st *s3Storage) Init(ctx context.Context, owner string, workspace string, instance string) error {
+	cfg, err := awsconfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	s3st.client = s3.NewFromConfig(cfg)
```

Comment:
nit: the client has been initialized at [here](https://github.com/gitpod-io/gitpod/blob/340817a3518c17bb1c4c9c045ac72a293606f007/components/content-service/pkg/storage/s3.go#L252-L255).